runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java

```diff
@@ -220,15 +220,18 @@ public DirectPipelineResult run(Pipeline pipeline) {
    */
   @SuppressWarnings("rawtypes")
   private List<PTransformOverride> defaultTransformOverrides() {
-    return ImmutableList.<PTransformOverride>builder()
-        .add(
-            PTransformOverride.of(
-                PTransformMatchers.writeWithRunnerDeterminedSharding(),
-                new WriteWithShardingFactory())) /* Uses a view internally. */
-        .add(
-            PTransformOverride.of(
-                PTransformMatchers.classEqualTo(CreatePCollectionView.class),
-                new ViewOverrideFactory())) /* Uses pardos and GBKs */
+    ImmutableList.Builder<PTransformOverride> builder =
+     ImmutableList.<PTransformOverride>builder();
+    if (!options.isUnitTest()) {
```

Comment:
If you add `DirectTestOptions` I believe you will have to register them in the `DirectPipelineRegistrar` as well